LocalFolder/explainer.md

```diff
@@ -0,0 +1,204 @@
+
+The proposed solution works by making use of the [Origin Private File System (OPFS)](https://web.dev/articles/origin-private-file-system) and the [File System Access](https://wicg.github.io/file-system-access/) APIs. It exposes an app's `LocalFolder` file system directory as an entry in the app origin's OPFS root directory.
+
+## Goals
+
+* Enable a Microsoft Store PWA to:
+   * read files from the `LocalFolder` directory specific to its PFN.
+   * delete any file or sub-directory in its `LocalFolder` directory. 
+* Avoid prompting end users to provide input or confirmation.
+* Avoid naming collisions with other OPFS root entries. 
+* Ensure that new behavior described here is only exposed to apps that opt in. 
+* Avoid introducing a new Web API that is only available on the Edge browser.
+
+## Non-goals
+
+* Enable a Microsoft Store PWA to create new files or modify existing files in its `LocalFolder` directory.
+* Support web apps not installed or managed by the Microsoft Store.
+* Support web apps on non-Windows platforms.
+* Standardization as a web specification.
+
+## Proposed Solution 
+
+### Existing behavior
+
+[`navigator.storage.getDirectory()`](https://developer.mozilla.org/en-US/docs/Web/API/StorageManager/getDirectory) is an existing API that returns a promise to a [FileSystemDirectoryHandle](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryHandle) which represents the root of directory the origin's OPFS storage space. By default, an origin's OPFS root directory has no entries.
+
+### New behavior
+
+When configured correctly, the directory handle for OPFS root will contain an additional entry that represents the app's `LocalFolder` directory. This entry can be retrieved as a `FileSystemDirectoryHandle` by calling [`.entries()`](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryHandle/entries) or [`.getDirectoryHandle(...)`](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryHandle/getDirectoryHandle) on the OPFS root directory handle.
+
+Like other OPFS handles, the `LocalFolder` directory handle and its contents will have [`readwrite`](https://wicg.github.io/file-system-access/#dom-filesystempermissionmode-readwrite) permission by default. Unlike other handles with `readwrite` permission, a [FileSystemFileHandle](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemFileHandle) from `LocalFolder` will always throw a [`NoModificationAllowedError`](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemFileHandle/createWritable#nomodificationallowederror) exception if [`createWritable()`](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemFileHandle/createWritable#exceptions) is used. This allows data in `LocalFolder` to be read and cleared but not created or edited.
```

Comment:
Won't you need to do something similar for `FileSystemDirectoryHandle` and `getDirectoryHandle()` so that attempts to create new sub-directories result in the same exception?

Reply:
Called out `getDirectoryHandle()` and `getFIleHandle()` with `create===true` as well.